Document LMStudio usage

**Describe the bug**
Trouble connecting to LMStudio

**Steps to Reproduce**
1.Start server on LMStudio
2.Start frontend and backend on OpenDevin
3.

**Expected behavior**
OpenDevin asks what I want it to build

**Actual behavior**

**Additional context**
<!-- Please include logs, screenshots, etc. -->
OpenDevin does nothing and I get this error in LMStudio:

[2024-03-30 19:21:30.994] [ERROR] Unexpected endpoint or method. (GET /litellm-models). Returning 200 anyway

> LLM_API_KEY="lm-studio"
> LLM_MODEL="openai/mistral" //leave openai as is... you can change mistral to the local model you use
> LLM_BASE_URL="http://localhost:1234/v1"
> LLM_EMBEDDING_MODEL="local"

this works but now OpenDevin wants to use /generate endoint instead of the ones we actually have available in LM-Studio

```bash
[2024-05-03 15:06:05.741] [INFO] [LM STUDIO SERVER] ->	GET  http://localhost:1234/v1/models
[2024-05-03 15:06:05.745] [INFO] [LM STUDIO SERVER] ->	POST http://localhost:1234/v1/chat/completions
[2024-05-03 15:06:05.748] [INFO] [LM STUDIO SERVER] ->	POST http://localhost:1234/v1/completions
[2024-05-03 15:06:05.752] [INFO] [LM STUDIO SERVER] ->	POST http://localhost:1234/v1/embeddings     <------------ NEW!
[2024-05-03 15:06:05.756] [INFO] [LM STUDIO SERVER] Logs are saved into D:\tmp\lmstudio-server-log.txt
[2024-05-03 18:13:56.284] [ERROR] Unexpected endpoint or method. (POST /v1/api/generate). Returning 200 anyway
[2024-05-03 18:13:58.040] [ERROR] Unexpected endpoint or method. (POST /v1/api/generate). Returning 200 anyway
[2024-05-03 18:13:59.117] [ERROR] Unexpected endpoint or method. (POST /v1/api/generate). Returning 200 anyway
[2024-05-03 18:14:00.676] [ERROR] Unexpected endpoint or method. (POST /v1/api/generate). Returning 200 anyway
[2024-05-03 18:14:03.072] [ERROR] Unexpected endpoint or method. (POST /v1/api/generate). Returning 200 anyway
```

--- a/docs/modules/usage/llms/localLLMs.md
+++ b/docs/modules/usage/llms/localLLMs.md
@@ -38,7 +38,7 @@ But when running `docker run`, you'll need to add a few more arguments:
 -e LLM_OLLAMA_BASE_URL="http://host.docker.internal:11434" \
 ```
 
-LLM_OLLAMA_BASE_URL is optional. If you set it, it will be used to show the available installed models in the UI. 
+LLM_OLLAMA_BASE_URL is optional. If you set it, it will be used to show the available installed models in the UI.
 
 For example:
 
@@ -53,6 +53,7 @@ docker run \
     -e SANDBOX_USER_ID=$(id -u) \
     -e LLM_API_KEY="ollama" \
     -e LLM_BASE_URL="http://host.docker.internal:11434" \
+    -e LLM_OLLAMA_BASE_URL="http://host.docker.internal:11434" \
     -e WORKSPACE_MOUNT_PATH=$WORKSPACE_BASE \
     -v $WORKSPACE_BASE:/opt/workspace_base \
     -v /var/run/docker.sock:/var/run/docker.sock \
@@ -68,12 +69,16 @@ Use the instructions in [Development.md](https://github.com/OpenDevin/OpenDevin/
 Make sure `config.toml` is there by running `make setup-config` which will create one for you. In `config.toml`, enter the followings:
 
 ```
-LLM_MODEL="ollama/codellama:7b"
-LLM_API_KEY="ollama"
-LLM_EMBEDDING_MODEL="local"
-LLM_BASE_URL="http://localhost:11434"
-WORKSPACE_BASE="./workspace"
-WORKSPACE_DIR="$(pwd)/workspace"
+[core]
+workspace_base="./workspace"
+
+[llm]
+model="ollama/codellama:7b"
+api_key="ollama"
+embedding_model="local"
+base_url="http://localhost:11434"
+ollama_base_url="http://localhost:11434"
+
 ```
 
 Replace `LLM_MODEL` of your choice if you need to.
@@ -141,4 +146,63 @@ Finally test that ollama is accessible from within the container
 ollama list # get list of installed models
 docker ps # get list of running docker containers, for most accurate test choose the open devin sandbox container.
 docker exec [CONTAINER ID] curl http://host.docker.internal:11434/api/generate -d '{"model":"[NAME]","prompt":"hi"}'
+```
+
+
+# Local LLM with LM Studio
+
+Steps to set up LM Studio:
+1. Open LM Studio
+2. Go to the Local Server tab.
+3. Click the "Start Server" button.
+4. Select the model you want to use from the dropdown.
+
+
+Set the following configs:
+```bash
+LLM_MODEL="openai/lmstudio"
+LLM_BASE_URL="http://localhost:1234/v1"
+CUSTOM_LLM_PROVIDER="openai"
+```
+
+### Docker
+
+```bash
+docker run \
+    -it \
+    --pull=always \
+    -e SANDBOX_USER_ID=$(id -u) \
+    -e LLM_MODEL="openai/lmstudio"
+    -e LLM_BASE_URL="http://host.docker.internal:1234/v1" \
+    -e CUSTOM_LLM_PROVIDER="openai"
+    -e WORKSPACE_MOUNT_PATH=$WORKSPACE_BASE \
+    -v $WORKSPACE_BASE:/opt/workspace_base \
+    -v /var/run/docker.sock:/var/run/docker.sock \
+    -p 3000:3000 \
+    ghcr.io/opendevin/opendevin:main
+```
+
+You should now be able to connect to `http://localhost:3000/`
+
+In the development environment, you can set the following configs in the `config.toml` file:
+
+```
+[core]
+workspace_base="./workspace"
+
+[llm]
+model="openai/lmstudio"
+base_url="http://localhost:1234/v1"
+custom_llm_provider="openai"
+```
+
+Done! Now you can start Devin by: `make run` without Docker. You now should be able to connect to `http://localhost:3000/`
+
+# Note:
+
+For WSL, run the following commands in cmd to set up the networking mode to mirrored:
+
+```
+python -c  "print('[wsl2]\nnetworkingMode=mirrored',file=open(r'%UserProfile%\.wslconfig','w'))"
+wsl --shutdown
 ```

